sycl/doc/OptionalDeviceFeatures.md

```diff
@@ -629,11 +625,126 @@ example, if a new device is released before there is a new DPC++ release.  In
+
+##### Aspect filter tool
+
+This tool transforms an input file table by removing rows with device code files
+that use features unsupported for the target architecture given as tool's
+argument.
+
+*Name*:
+
+- `sycl-aspect-filter`, located next to other tools like `file-table-tform`
+
+*Input*:
+
+- file table, normally coming out of `sycl-post-link` or `file-table-tform`
+  tools
+
+*Command line arguments*:
+
+- `-target=<target>` target device architecture to filter for
+- `-device-config-file=<path>` path to the device configuration file
+
+*Output*:
+
+- the input file table filtered as needed
+
+In more details, the tool performs the following actions:
+
+1) Checks if the input file table contains "Properties" column. If not, copies
+   the input file table to output and exits without error.
+1) Reads in the device configuration file and finds some entry `E` corresponding
+   to the architecture given on the command line. If there is no such entry -
+   reports an error and exits.
+1) For each row in the input file table:
+
+   - loads the properties file from the "Properties" column
+   - checks if there is the `SYCL/device-requirements` property
+   - if no, copies current row to the output file table and goes to the next
+   - if yes, checks if all the requirements listed in the property are supported
+     by the target architecture as specified by entry `E` in the device
+     configuration file
+       - if yes, copies current row to the output file table and goes to the
+         next
+       - otherwise skips this row
+
+##### Configuration file location and driver option
+
+A default device configuration file is present in DPC++ build. Users may override it using the
+`-fsycl-device-config-file=<path>` compiler command line option.
+Exact location of the file and final name of the compiler option is TBD.
+
+##### AOT target identification
+
+There are several user-visible places in the SDK where SYCL device target
+architectures need to be identified:
+
+- `-fsycl-targets` option
+- a device configuration file entry
+- `-target` option of the `sycl-aspec-filter` tool
+- a SYCL aspect enum identifier (we expect to add a new SYCL aspect for each
+  device target architecture)
+
+In all such places architecture naming should be the same. In some cases aliases
+are allowed. Below is a list of target architectures supported by DPC++:
+
+| target/alias(es) | description |
+|-|-|
+| intel_gpu    | Generic Intel graphics architecture |
+| intel_gpu_tgl, intel_gpu_12_0 | Intel Tiger Lake (11th generation Core) integrated graphics architecture |
+| ptx64  | Generic 64-bit PTX target architecture |
+| spir64 | Generic 64-bit SPIR-V target |
+| x86_64 | Generic 64-bit x86 architecture |
+
+TODO: Provide full list of AOT targets supported by the identification
+mechanism.
+
+Example of clang compilation invocation with 2 AOT targets and generic SPIR-V:
+```
+clang++ -fsycl -fsycl-targets=spir64,intel_gpu_12_0,ptx64 ...
```

Comment:
A belated note: this example is not correct, since the `-fsycl-targets=` option doesn't support Intel Graphics platform specification (that is done via `-Xsycl-target-backend "-device ..."` mechanism of option passthrough to `ocloc`). The closest a compiler user would be able to get to this would be:
```
clang++ -fsycl -fsycl-targets=spir64,spir64_gen,nvptx64-nvidia-cuda -Xsycl-target-backend=spir64_gen "-device gen12" ...
```
However, I realize that some interface changes may be planned in this regard that are not covered by docs for option device features. So what would be the correct way to reformulate this bit, given the planned compiler architecture changes?